File: src/VBNetFx/FAC_Generic/FacessoApplicationSettings.vb
Imports System.Configuration

Public Class FacessoApplicationSettings
    Inherits ApplicationSettingsBase

    Sub New()
        MyBase.New("FacessoGlobal")
    End Sub

    <Global.System.Configuration.UserScopedSettingAttribute()> _
    Public Property LoginHistory() As LoginHistory
        Get
            Return CType(Me("LoginHistory"), LoginHistory)
        End Get
        Set(ByVal value As LoginHistory)
            Me("LoginHistory") = value
        End Set
    End Property

    <Global.System.Configuration.UserScopedSettingAttribute()> _
    Public Property LastLoginName() As String
        Get
            Return CType(Me("LastLoginName"), String)
        End Get
        Set(ByVal value As String)
            Me("LastLoginName") = value
        End Set
    End Property

    <Global.System.Configuration.UserScopedSettingAttribute()> _
    Public Property LastSubsidiaryID() As Integer
        Get
            Return CType(Me("LastSubsidiaryID"), Integer)
        End Get
        Set(ByVal value As Integer)
            Me("LastSubsidiaryID") = value
        End Set
    End Property
End Class

<Serializable()> _
Public Class FacessoDynamicSettingsList
    Inherits System.Collections.Hashtable

    Public Function GetItem(ByVal Key As String, ByVal DefaultValue As Object) As Object
        If Me.ContainsKey(Key) Then
            Return Me(Key)
        End If
        Me.Add(Key, DefaultValue)
        Return DefaultValue
    End Function

    Public Sub SetItem(ByVal key As String, ByVal Value As Object)
        If Me.ContainsKey(key) Then
            Me(key) = Value
            Return
        End If
        Me.Add(key, Value)
    End Sub
End Class

File: src/MixNetFx/FAC_Generic/FacessoApplicationSettings.cs
using System.Configuration;

namespace Facesso
{
    public class FacessoApplicationSettings : ApplicationSettingsBase
    {
        public FacessoApplicationSettings() : base("FacessoGlobal") { }

        [UserScopedSetting]
        public LoginHistory LoginHistory
        {
            get { return (LoginHistory)this["LoginHistory"]; }
            set { this["LoginHistory"] = value; }
        }

        [UserScopedSetting]
        public string LastLoginName
        {
            get { return (string)this["LastLoginName"]; }
            set { this["LastLoginName"] = value; }
        }

        [UserScopedSetting]
        public int LastSubsidiaryID
        {
            get { return (int)this["LastSubsidiaryID"]; }
            set { this["LastSubsidiaryID"] = value; }
        }
    }

    [System.Serializable]
    public class FacessoDynamicSettingsList : System.Collections.Hashtable
    {
        public object GetItem(string key, object defaultValue)
        {
            if (ContainsKey(key))
                return this[key];
            Add(key, defaultValue);
            return defaultValue;
        }

        public void SetItem(string key, object value)
        {
            if (ContainsKey(key))
            {
                this[key] = value;
                return;
            }
            Add(key, value);
        }
    }
}
